src/GraphQL/Execution/ExecutionNode.cs

```diff
@@ -59,6 +59,46 @@ public IObjectGraphType GetParentType(ISchema schema)
 
             return null;
         }
+
+        public IEnumerable<string> Path
+        {
+            get
+            {
+                var pathList = new List<string>();
```

Comment:
Have you tried looking at the performance numbers of this method, if you determine the size of the list in advance and create it with the desired capacity? It requires do `while` loop twice, one time to count desired size, second time to fill the list, but it can save some allocations, so...

Reply:
No but I could.  Should be pretty easy to try

Reply:
It will be interesting to compare the results.

Reply:
Uhh, well, it's easy to write the code anyway.  But Path isn't used by the standard benchmarks, so none of this code typically runs.

Reply:
It also eliminates the call to Reverse

Reply:
```c#
  private ExecutionResult ExecuteQuery(ISchema schema, string query)
        {
            return _executer.ExecuteAsync(_ =>
            {
                _.Schema = schema;
                _.Query = query;
                _.EnableMetrics = true;       USE THIS TO FORCE ACCESS TO PATH
            }).GetAwaiter().GetResult();
        }
```

Reply:
Take a look.  I didn't run benchmarks, but it's pretty obvious it should run faster, when there's no List object to allocate, and no Reverse() call -- just the raw array

Reply:
By the way, I think that it makes no sense to make an iterator inside `Path`, because if there is a call to this property, then the consumer will read all the elements. By default, no one needs this property. It is used in advanced scenarios.